src/playfair.rs

```diff
@@ -0,0 +1,406 @@
+//! The Playfair cipher is the first bigram substitution cipher.
+//! Invented in 1854 by Charles Wheatstone, its name honors "Lord"
+//! Lyon Playfair for promoting its use.
+//!
+//! [Reference](https://en.wikipedia.org/wiki/Playfair_cipher)
+//!
+//! The Playfair cipher operates on a 5x5 table. The key, omitting repeated
+//! characters, is written from left to right starting on the first row
+//! of the table. Other key layout patterns in the table can be used
+//! but are less common. Note that a letter must either be omitted
+//! (typically 'Q') or two letters can occupy the same space (I=J).
+//! This implementation uses the *latter* design, replacing all
+//! encountered 'J' characters with 'I'.
+//!
+use common::{alphabet, alphabet::Alphabet, cipher::Cipher, keygen::PlayfairTable};
+
+/// The character inserted to avoid repeated characters or
+/// to complete an odd-length bigram
+const PLAYFAIR_FIX_CHAR: char = 'X';
+
+/// A Playfair cipher.
+///
+/// This struct is created by the `new()` method. See its documentation for more.
+pub struct Playfair {
+    /// The Playfair key table (5x5)
+    table: PlayfairTable,
+}
+
+impl Cipher for Playfair {
+    type Key = String;
+    type Algorithm = Playfair;
+
+    /// Initialize a Playfair cipher.
+    ///
+    /// # Warning
+    /// The 5x5 key table requires any 'J' characters in the key
+    /// to be substituted with 'I' characters (I = J).
+    fn new(key: Self::Key) -> Result<Playfair, &'static str> {
+        let key_table = PlayfairTable::new(&key)?;
+
+        Ok(Playfair { table: key_table })
+    }
+
+    /// Encrypt a message with the Playfair cipher.
+    ///
+    /// Accepts messages consisting only of alpha characters and whitespace.
+    /// The resulting plaintext will be fully uppercase with no spaces.
+    ///
+    /// # Examples
+    ///
+    /// Basic Usage:
+    ///
+    /// ```
+    /// use cipher_crypt::{Cipher, Playfair};
+    ///
+    /// let c = Playfair::new("playfair example".to_string()).unwrap();
+    /// assert_eq!(
+    ///     c.encrypt("Hide the gold in the tree stump").unwrap(),
+    ///     "BMODZBXDNABEKUDMUIXMMOUVIF"
+    /// );
+    /// ```
+    ///
+    /// # Warning
+    ///
+    /// * The 5x5 key table requires any 'J' characters in the message
+    /// to be substituted with 'I' characters (i.e. I = J).
+    /// * The resulting ciphertext will be fully uppercase with no whitespace.
+    fn encrypt(&self, message: &str) -> Result<String, &'static str> {
+        let message: String = message.split_whitespace().collect();
+        if !alphabet::STANDARD.is_valid(message.as_str()) {
+            return Err("Message must only consist of alphabetic characters");
+        }
+
+        // Handles Rule 1
```

Comment:
Unless I've overlooked it somewhere in the documentation, what is Rule 1? If you've already explained this rule, make sure to label it as rule 1.